### ISSUE TITLE
Cannot add InAppMessaging to project through cocoapods

### ISSUE THREAD

**Issue by barryki** [firebase/firebase-ios-sdk#11569]
### Description

If platform is tvos, Podfile.lock and after running pod install, InAppMessaging appeared in PODS and DEPENDENCIES sections, but missing from SPEC REPOS and SPEC CHECKSUMS. Pods folder does not have the module. It works fine if platform is ios.

### Reproducing the issue

- Create a new tvOS project with XCode
- Add Podfile
<details>
<summary>Expand <code>Podfile</code> snippet</summary>
<br>

```yml

project 'Cocoapods Testing.xcodeproj'
platform :tvos, '12.0'

source 'https://github.com/CocoaPods/Specs.git'

target 'Cocoapods Testing' do
  inhibit_all_warnings!
  use_frameworks!

  pod 'Firebase/InAppMessaging', '~> 10.0'

end


```

</details>
- Run `pod install` in terminal
- Check Podfile.lock

### Firebase SDK Version

10.12

### Xcode Version

14.1

### Installation Method

CocoaPods

### Firebase Product(s)

In-App Messaging

### Targeted Platforms

tvOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

PODS:
  - Firebase/CoreOnly (10.12.0):
    - FirebaseCore (= 10.12.0)
  - Firebase/InAppMessaging (10.12.0):
    - Firebase/CoreOnly
    - FirebaseInAppMessaging (~> 10.12.0-beta)
  - FirebaseCore (10.12.0):
    - FirebaseCoreInternal (~> 10.0)
    - GoogleUtilities/Environment (~> 7.8)
    - GoogleUtilities/Logger (~> 7.8)
  - FirebaseCoreInternal (10.12.0):
    - "GoogleUtilities/NSData+zlib (~> 7.8)"
  - GoogleUtilities/Environment (7.11.1):
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleUtilities/Logger (7.11.1):
    - GoogleUtilities/Environment
  - "GoogleUtilities/NSData+zlib (7.11.1)"
  - PromisesObjC (2.2.0)

DEPENDENCIES:
  - Firebase/InAppMessaging (~> 10.0)

SPEC REPOS:
  https://github.com/CocoaPods/Specs.git:
    - Firebase
    - FirebaseCore
    - FirebaseCoreInternal
    - GoogleUtilities
    - PromisesObjC

SPEC CHECKSUMS:
  Firebase: 07150e75d142fb9399f6777fa56a187b17f833a0
  FirebaseCore: f86a1394906b97ac445ae49c92552a9425831bed
  FirebaseCoreInternal: 950500ad8a08963657f6d8c67b579740c06d6aa1
  GoogleUtilities: 9aa0ad5a7bc171f8bae016300bfcfa3fb8425749
  PromisesObjC: 09985d6d70fbe7878040aa746d78236e6946d2ef

PODFILE CHECKSUM: b5a09a73a41c135f1805ea2fc54e2abeb535e576

COCOAPODS: 1.12.1


```

</details>

**Comment by paulb777** [comment]
@barryki Thanks for the report and detailed repro instructions. There was an issue in the Firebase podspec that will be fixed in Firebase 10.13.0.

In the meantime, a workaround is to explicitly install In App Messaging by adding the following to the Podfile:  ` pod 'FirebaseInAppMessaging', '~> 10.0-beta'`

**Comment by barryki** [comment]
> In the meantime, a workaround is to explicitly install In App Messaging by adding the following to the Podfile

Great. Thanks!
